### ISSUE TITLE
Player

### ISSUE THREAD

**Issue by nt4f04uNd** [florent37/Flutter-AssetsAudioPlayer#426]
**Flutter Version**
1.24.0-8.0.pre.374

**Lib Version**

2.0.10

**Platform**

iOS 14.01

**Describe the bug**

Start player -> start some other audio source -> player stops and I cannot update my UI to respect that
On android it will keep playing

Might be helpful
https://github.com/flutter/flutter/issues/66799

**Comment by nt4f04uNd** [comment]
ok i found out `isPlaying`, gonna test this
described behaviour is still not intended though, i want to keep the player playing

**Comment by nt4f04uNd** [comment]
this is happening with Youtube app in particular, so i guess probably there's nothing you can do here and the issue should be closed?

**Comment by kalismeras61** [comment]
Are you having problem with only IOS ?

**Comment by nt4f04uNd** [comment]
yes, this is only related to darwin and here's why - i found that when you don't `respectSilence`, the audio session category is forced to `AVAudioSessionCategoryPlayback`, which also makes that controls appear on the lockscreen no matter any other parameters

https://github.com/florent37/Flutter-AssetsAudioPlayer/blob/89ae26204d7ca312208b3bacb29b68ebbfe61b1c/darwin/Classes/Music.swift#L178-L183

as far a i know this is a general limitation on darwin, you:
* either set you session to be ambient and thus have playback that is not interrupted by other apps, but this session will be muted by the silence switcher
* or have other type of session which force the controls to appear, they interrupt other sessions and do not respect the silence switcher

i might be wrong as there are a few other types of [categories](https://developer.apple.com/documentation/avfaudio/avaudiosessioncategory), which also have some [options](https://developer.apple.com/documentation/avfaudio/avaudiosessioncategoryoptions?language=objc)

so if you are going to work on this issue, i encourage to investigate all this thing, and rework the current package API in that part, because it makes an illusion that it will work, whilst it actually won't. i'm talking about `AudioFocusStrategy` (mixing with others), respecting silence and showing media controls (and might be something else idk). also, all the audio session categories should be supported

**Comment by kalismeras61** [comment]
I did some changes . Can you try below branch ? 
  ```
assets_audio_player:
    git:
      url: https://github.com/florent37/Flutter-AssetsAudioPlayer.git
      ref: fix_audiofocus
```

**Comment by nt4f04uNd** [comment]
i think it would be better if you could do this yourself, but if you really need me to do this, i'm gonna tomorrow/day after

**Comment by kalismeras61** [comment]
@nt4f04uNd i already tried but, As the person reporting the problem, you have to test it.

**Comment by nt4f04uNd** [comment]
@kalismeras61 sorry for a delayed response - no, AudioFocusStrategy.none() still doesn't work (tested in my app on iPhone 7 iOS 14.3)

if you can provide a reproducible build w/ and w/o fix and share them with me via https://www.installonair.com/home i could test it for you

**Comment by uugurbas** [comment]
any solutions? I am having the same problem. When trying to play assets several times consecutively it stops playing.
